docs/chapters/incremental_indexing/index.md

```diff
@@ -0,0 +1,68 @@
+# Incremental Indexing in Jina
+
+## Generating Documents Ids:
+
+In Jina, the document ID is by default generated deterministically by computing a hash the document content. This avoids repeating the doc id while reindexing by using a Random or Incremental counter. The user can `override` document ids with Jina client and add `tags` to map to their unique concepts.
+A customized id is only acceptable when: it only contains the symbols "0"–"9" to represent values 0 to 9, and "A"–"F" (or alternatively "a"–"f") it has an even length.'
+
+## Incremental Indexing:
+
+The rationale behind Incremental indexing is to avoid re-indexing previously indexed documents, thus saving time and compute. This is done by persisting the document identifiers of indexed docs and using them as a DocIdCache.
+If user does not expect to send the same document again, it can work without adding the extra caches, but if they want to avoid duplicating documents, they could use incremental indexing patterns.
+While indexing a document, it’s identifier is checked against the Cache for a ‘hit’ - if the document id exists,it’s already been indexed. We skip re-indexing it. If it’s a ‘miss’, we index it, adding the identifier to the Cache.
+1. *DocIdCache* stores doc ids in a int64 set and persists it to a numpy array
+2. *UniqueVectorIndexer* combines VectorIndexer with DocIdCache
+```yaml
+!UniqueVectorIndexer
+components:
+  - !DocIDCache
+    metas:
+      workspace: $JINA_TEST_INCREMENTAL_INDEX_WORKSPACE
```

Comment:
Edit the env variable to $WORKSPACE?

Reply:
Actually, no need to add the metas, I think the point is to showcase the concept, abd adding metas and this adds noise